secant/Util/RecoveryPhraseStorage.swift

```diff
@@ -0,0 +1,228 @@
+            
+            try setData(data, forKey: Constants.zcashBirthday)
+        } catch {
+            throw error
+        }
+    }
+    
+    /// Restore data for key
+    func data(
+        forKey: String,
+        account: String = ""
+    ) -> Data? {
+        let query = RecoveryPhraseStorage.restoreQuery(forAccount: account, andKey: forKey)
+
+        var result: AnyObject?
+        _ = SecItemCopyMatching(query as CFDictionary, &result)
+        
+        return result as? Data
+    }
+    
+    /// Use carefully:  Deletes data for key
+    func deleteData(
+        forKey: String,
+        account: String = ""
+    ) -> Bool {
+        let query = RecoveryPhraseStorage.baseQuery(forAccount: account, andKey: forKey)
+
+        let status = SecItemDelete(query as CFDictionary)
+
+        return status == noErr
+    }
+    
+    /// Store data for key
+    func setData(
+        _ data: Data,
+        forKey: String,
+        account: String = ""
+    ) throws {
+        var query = RecoveryPhraseStorage.baseQuery(forAccount: account, andKey: forKey)
+        query[kSecValueData as String] = data as AnyObject
+
+        let status = SecItemAdd(query as CFDictionary, nil)
+        
+        guard status != errSecDuplicateItem else {
+            throw KeychainError.duplicate
+        }
+        
+        guard status == errSecSuccess else {
+            throw KeychainError.unknown(status)
+        }
+    }
+}
+
+// MARK: - KeyStoring
+
+extension RecoveryPhraseStorage: KeyStoring {
+    func importBirthday(_ height: BlockHeight) throws {
+        do {
+            try setBirthday(height)
+        } catch RecoveryPhraseStorage.KeychainError.duplicate {
+            throw KeyStoringError.alreadyImported
+        } catch {
+            throw KeyStoringError.storageError(error)
+        }
+    }
+    
+    func exportBirthday() throws -> BlockHeight {
+        do {
+            return try birthday()
+        } catch {
+            throw KeyStoringError.uninitializedWallet
+        }
+    }
+    
+    func importPhrase(bip39 phrase: String) throws {
+        do {
+            try setRecoveryPhrase(phrase)
+        } catch RecoveryPhraseStorage.KeychainError.duplicate {
+            throw KeyStoringError.alreadyImported
+        } catch {
+            throw KeyStoringError.storageError(error)
+        }
+    }
+    
+    func exportPhrase() throws -> String {
+        do {
+            return try recoveryPhrase()
+        } catch {
+            throw KeyStoringError.uninitializedWallet
+        }
+    }
+    
+    func areKeysPresent() throws -> Bool {
+        do {
+            _ = try exportPhrase()
+            _ = try exportBirthday()
+        } catch KeyStoringError.uninitializedWallet {
+            return false
+        } catch {
+            // TODO: - report & log error.localizedDescription
+            return false
+        }
+        
+        return true
+    }
+    
+    func nukePhrase() {
```

Comment:
Do we ever expect callers to call delete individually?  I wonder whether these individual methods should be `private`?

Reply:
solved, nukeWallet() is the only function now